pkg/v3/store/metadata_store.go

```diff
@@ -0,0 +1,162 @@
+package store
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/smartcontractkit/ocr2keepers/pkg/util"
+	"github.com/smartcontractkit/ocr2keepers/pkg/v3/tickers"
+	"github.com/smartcontractkit/ocr2keepers/pkg/v3/types"
+)
+
+const (
+	proposalLogExpiry          = 24 * time.Hour
+	proposalLogCleanupInterval = time.Hour
+)
+
+//go:generate mockery --name MetadataStore --structname MockMetadataStore --srcpkg "github.com/smartcontractkit/ocr2keepers/pkg/v3/store" --case underscore --filename metadatastore.generated.go
+type MetadataStore interface {
+	SetBlockHistory(blockHistory types.BlockHistory)
+	GetBlockHistory() types.BlockHistory
+	GetProposalLogRecovery(key string) (types.CoordinatedProposal, bool)
+	SetProposalLogRecovery(key string, value types.CoordinatedProposal, expire time.Duration)
+	GetProposalLogRecoveryKeys() []string
+	RemoveProposalLogRecovery(key string)
+	ClearAllProposalLogRecovery()
+	ClearExpiredProposalLogRecovery()
+	AppendProposalConditional(...types.CoordinatedProposal)
+	GetProposalConditional() []types.CoordinatedProposal
+	RemoveProposalConditional(...types.CoordinatedProposal) []types.CoordinatedProposal
+	Start(context.Context) error
+	Close() error
+}
+
+type metadataStore struct {
+	blocks                     *tickers.BlockTicker
+	blockHistory               types.BlockHistory
+	blockHistoryMutex          sync.RWMutex
+	proposalLogRecovery        *util.Cache[types.CoordinatedProposal]
```

Comment:
This was pretty surprising; the value stored for `ProposalRecoveryMetadata` wasn't a list of objects, but a separate cache? I also couldn't find a cache cleaner for this cache, so I'm not sure if that could have potentially caused a memory leak? 

Beyond retrieving the cache, there didn't seem to be use of a mutex when reading/updating values, so I'm not sure if that's something we need to consider here or not?